sdk/formrecognizer/ai-form-recognizer/review/ai-form-recognizer.api.md

```diff
@@ -0,0 +1,1299 @@
+
+// @public
+export interface CopyAuthorization {
+    accessToken: string;
+    expirationDateTime: Date;
+    targetModelId: string;
+    targetModelLocation: string;
+    targetResourceId: string;
+    targetResourceRegion: string;
+}
+
+// @public
+export interface CopyModelOptions extends OperationOptions, PollerOptions<TrainingPollOperationState> {
+}
+
+// @public
+export interface CustomDocumentModelsInfo {
+    count: number;
+    limit: number;
+}
+
+// @public
+export interface DateFieldSchema {
+    // (undocumented)
+    readonly type: "date";
+}
+
+// @public
+export interface DeleteModelOptions extends OperationOptions {
+}
+
+// @public
+export interface DocTypeInfo {
+    description?: string;
+    fieldConfidence?: {
+        [propertyName: string]: number;
+    };
+    fieldSchema: {
+        [propertyName: string]: DocumentFieldSchema;
+    };
+}
+
+// @public
+export class DocumentAnalysisClient {
+    constructor(endpoint: string, credential: TokenCredential, options?: DocumentAnalysisClientOptions);
+    constructor(endpoint: string, credential: KeyCredential, options?: DocumentAnalysisClientOptions);
+    // @internal
+    constructor(endpoint: string, credential: KeyCredential | TokenCredential, options?: DocumentAnalysisClientOptions);
+    beginAnalyzeDocuments(modelId: string, input: string | FormRecognizerRequestBody, options?: AnalyzeDocumentsOptions): Promise<AnalysisPoller>;
+    beginAnalyzeDocuments<Document>(model: DocumentModel<Document>, input: string | FormRecognizerRequestBody, options?: AnalyzeDocumentsOptions<AnalyzeResult<Document>>): Promise<AnalysisPoller<AnalyzeResult<Document>>>;
+    beginExtractGenericDocument(input: string | FormRecognizerRequestBody, options?: AnalyzeDocumentsOptions<GenericDocumentResult>): Promise<AnalysisPoller<GenericDocumentResult>>;
+    beginExtractLayout(input: string | FormRecognizerRequestBody, options?: AnalyzeDocumentsOptions<LayoutResult>): Promise<AnalysisPoller<LayoutResult>>;
+}
+
+// @public
+export interface DocumentAnalysisClientOptions extends CommonClientOptions {
+}
+
+// @public
+export interface DocumentAnalysisPollOperationState<Result = AnalyzeResult<AnalyzedDocument>> extends PollOperationState<Result> {
+    modelId: string;
+    operationId: string;
+    status: AnalyzeResultOperationStatus;
+}
+
+// @public
+export interface DocumentArrayField<T = DocumentField> extends DocumentFieldCommon {
+    // (undocumented)
+    kind: "array";
+    values: T[];
+}
+
+// @public
+export interface DocumentCountryRegionField extends DocumentFieldCommon {
+    // (undocumented)
+    kind: "countryRegion";
+    value?: string;
+}
+
+// @public
+export interface DocumentCurrencyField extends DocumentFieldCommon {
+    // (undocumented)
+    kind: "currency";
+    value?: string;
+}
+
+// @public
+export interface DocumentDateField extends DocumentValueField<Date> {
+    // (undocumented)
+    kind: "date";
```

Comment:
undocumented property

Reply:
I can add a docstring to these kind/docType fields, but they will all say something like:

```javascript
/**
 * The kind of the DocumentField - "date".
 */
 ```
 And the user will never see it, because they would never have any reason to access the `kind` property in any context where its value could be known precisely.
 
 Maybe I could add `kind: string;` with a docstring to `DocumentFieldCommon`, but I don't know how that will behave with the rigid type union of the full DocumentField.

Reply:
I see. I am fine leaving `kind` undocumented. However please double check all other occurrences.